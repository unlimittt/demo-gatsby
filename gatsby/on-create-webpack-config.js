const onCreateWebpackConfig = ({ stage, actions }) => {
  if (stage.startsWith("develop")) {
    actions.setWebpackConfig({
      resolve: {
        alias: {
          "react-dom": "@hot-loader/react-dom",
        },
      },
    })
  }
}

module.exports = onCreateWebpackConfig
