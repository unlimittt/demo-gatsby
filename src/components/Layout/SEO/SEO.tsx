import React from "react"
import { Helmet } from "react-helmet"
import { useSiteMetadata } from "../../../hooks"
import { withPrefix } from "gatsby"

type Props = {
  title: string
  description?: string
  lang?: string
  meta?: []
  socialImage?: string
}

function SEO({
  title,
  description,
  lang = "en",
  meta = [],
  socialImage,
}: Props) {
  const siteMetadata = useSiteMetadata()

  const metaImage =
    socialImage != null ? socialImage : siteMetadata.author.photo
  const metaImageUrl = siteMetadata.url + withPrefix(metaImage)
  const metaDescription = description || siteMetadata.subtitle

  return (
    <Helmet
      htmlAttributes={{
        lang,
      }}
      title={title}
      titleTemplate={`%s - ${siteMetadata.title}`}
      meta={[
        {
          name: `description`,
          content: metaDescription,
        },
        {
          property: `og:title`,
          content: title,
        },
        {
          property: `og:description`,
          content: metaDescription,
        },
        {
          property: `og:type`,
          content: `website`,
        },
        {
          property: `og:site_name`,
          content: title,
        },
        {
          property: `og:image`,
          content: metaImageUrl,
        },
        {
          name: `twitter:card`,
          content: `summary`,
        },
        {
          name: `twitter:creator`,
          content: siteMetadata.author.contacts.twitter,
        },
        {
          name: `twitter:title`,
          content: title,
        },
        {
          name: `twitter:description`,
          content: metaDescription,
        },
        {
          name: `twitter:image`,
          content: metaImageUrl,
        },
      ].concat(meta)}
    />
  )
}

export default SEO
