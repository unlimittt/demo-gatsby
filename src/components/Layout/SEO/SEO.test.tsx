import React  from "react"
import { useStaticQuery, StaticQuery } from 'gatsby';
import siteMetadata from '../../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../../types';
import Enzyme, { shallow } from 'enzyme';
import Adapter from "enzyme-adapter-react-16"
import SEO from "./SEO";

Enzyme.configure({adapter: new Adapter()})

describe('SEO', () => {
  const props = {
    ...siteMetadata,
    title: 'test',
    description: 'test',
    lang: 'ru'
  };

  beforeEach(() => {
    (StaticQuery as jest.MockedFunction<typeof StaticQuery | any>).mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(props)
      ),
      (useStaticQuery as jest.MockedFunction<typeof useStaticQuery | any>).mockReturnValue(props)
    );
  });

  it('renders correctly', () => {
    const wrapper = shallow(<SEO {...props} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('HelmetWrapper').prop('htmlAttributes')).toEqual({lang: "ru"})
    expect(wrapper.find('HelmetWrapper').prop('meta')).toContainEqual({
      property: "og:image",
      content: "http://localhost/test.jpg",
    })
  });
});
