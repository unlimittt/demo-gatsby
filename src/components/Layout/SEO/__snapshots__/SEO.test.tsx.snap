// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO renders correctly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "lang": "ru",
    }
  }
  meta={
    Array [
      Object {
        "content": "test",
        "name": "description",
      },
      Object {
        "content": "test",
        "property": "og:title",
      },
      Object {
        "content": "test",
        "property": "og:description",
      },
      Object {
        "content": "website",
        "property": "og:type",
      },
      Object {
        "content": "test",
        "property": "og:site_name",
      },
      Object {
        "content": "http://localhost/test.jpg",
        "property": "og:image",
      },
      Object {
        "content": "summary",
        "name": "twitter:card",
      },
      Object {
        "content": "#",
        "name": "twitter:creator",
      },
      Object {
        "content": "test",
        "name": "twitter:title",
      },
      Object {
        "content": "test",
        "name": "twitter:description",
      },
      Object {
        "content": "http://localhost/test.jpg",
        "name": "twitter:image",
      },
    ]
  }
  title="test"
  titleTemplate="%s - Test title"
/>
`;
